modules/EvseManager/energy_grid/energyImpl.cpp

```diff
@@ -125,10 +139,20 @@ void energyImpl::request_energy_from_energy_manager(bool priority_request) {
         charger_state == Charger::EvseState::WaitingForAuthentication ||
         charger_state == Charger::EvseState::ChargingPausedEV || !mod->config.request_zero_power_in_idle) {
 
-        // copy complete external limit schedules
+        // copy complete external limit schedules for import
         if (mod->getLocalEnergyLimits().schedule_import.has_value() &&
             !mod->getLocalEnergyLimits().schedule_import.value().empty()) {
             energy_flow_request.schedule_import = mod->getLocalEnergyLimits().schedule_import;
+
+            if (mod->config.charge_mode == "DC") {
+                // For DC, apply our power supply capabilities as an additional limit on leaves side
+                const auto caps = mod->get_powersupply_capabilities();
+                for (auto& entry : energy_flow_request.schedule_import.value()) {
+                    if (entry.limits_to_leaves.total_power_W > caps.max_export_power_W) {
```

Comment:
Is export correct here?

Reply:
export on dc power supply is into the car, which is import from grid